doc/source/development/rfc/rfc100_float16_support.rst

```diff
@@ -0,0 +1,132 @@
+.. _rfc-100:
+
+=============================
+RFC 100: Support float16 type
+=============================
+
+============== =============================================
+Author:        Erik Schnetter
+Contact:       eschnetter @ perimeterinstitute.ca
+Started:       2024-Jun-05
+Status:        Proposed
+Target:        GDAL 3.10
+============== =============================================
+
+Summary
+-------
+
+This RFC adds support for the IEEE 16-bit floating point data type
+(aka ``half``, ``float16``). It adds a new pixel data type
+``GDT_Float16``. Where supported by the compiler, it also adds a new
+C/C++ type alias ``GFloat16``. Even when the local system does not
+support ``GFloat16``, datasets of this type can still be accessed by
+converting from/to ``float``.
+
+Some drivers, in particular Zarr, will be extended to support this
+type. Drivers that do not support this type will be checked to ensure
+they report an appropriate errors when necessary.
+
+Motivation
+----------
+
+- The float16 type is increasingly common. It is supported by GPUs, by
+  modern CPUs, and by modern compilers (e.g. GCC 12).
+
+- Using ``uint16`` values to shuffle ``float16`` values around is
+  inconvenient and awkward. There is no mechanism in GDAL to mark
+  attributes or datasets which ``uint16`` values should be interpreted
+  as ``float16``.
+
+- C++23 will introduce native support for ``std::float16_t``. However,
+  it will likely be several years until C++23 will be a requirement
+  for GDAL. A shorter-term solution is needed.
+
+- Many other projects and storage libraries support float16. GDAL will
+  fall behind if it doesn't.
+
+Details
+-------
+
+The following type aliases are defined in both C and C++, if supported
+by the compiler:
+- ``GFloat16`` --> e.g. ``_Float16``
+
+The preprocessor symbol ``GDAL_HAVE_GFLOAT16`` indicates whether
+``GFloat16`` is available at compile time. This information will also
+be available at run time via ``bool GDALHaveFloat16()``.
+
+The following pixel data types are added:
+- ``GDT_Float16``  --> ``GFloat16``
+- ``GDT_CFloat32`` --> ``std::complex<GFloat16`` / ``_Complex GFloat16``
+
+These enum values are added independent of the value of whether the
+compiler supports float16.
+
+For simplicity there are no attribute functions for float16 values.
+Attributes can still be read/written as raw values.
```

Comment:
I assume you are speaking about the multidimensional API here? Should be precised